[xla hash update] update the pinned xla hash

ezyang opened the pull request:
This PR is auto-generated nightly by [this action](https://github.com/pytorch/pytorch/blob/master/.github/workflows/_update-commit-hash.yml).
Update the pinned xla hash.

zdevito commented on torch/csrc/jit/ir.h:
```diff
@@ -30,7 +33,18 @@ struct Graph;
+  Single
+};
+
+struct Type {
```
I think we should make Type into a class hierarchy like Node*. It is more future proof. Here we already have to types: TensorType and MultiReturnType.

apaszke replied:
Ok, sounds good

zdevito commented on torch/csrc/jit/ir.h:
```diff
@@ -86,22 +100,33 @@ struct Node {
+  const Type & type() {
     return type_;
   }
+  void registerOutput(thpp::Tensor* output) {
```
Maybe name "inferNodeTypeFrom("

zdevito commented on torch/csrc/jit/ir.h:
```diff
@@ -355,12 +380,13 @@ struct Node {
 // without significant boilerplate
 
-template<typename Self, NodeKind K>
+template<typename Self, NodeKind K, ReturnKind R>
```
I think this stuff will be cleaned up if we handle multi-return as part of Type.

apaszke replied:
Can you elaborate? I'm not sure how could we change this (we at least need to know what kind of Type to instantiate for a node)

zdevito replied:
Yes, we'd need to keep this. We should rename `NodeWithKind` because now it is handling 3 different things: Kind, Clone, and the kind of Type. Is there a C++ naming convention for this kind of CRTP object?

apaszke replied:
I don't know 😕 `NodeBase`?

zdevito replied:
Well, if it doesn't have a standard name, then we leave it as is...

zdevito commented on torch/csrc/jit/ir.h:
```diff
@@ -379,29 +489,28 @@ struct NodeWithKind : public Node {
 // helper to define simple primitive Ops.
-template<typename Self, NodeKind K>
-struct Primitive : public NodeWithKind<Self,K> {
+template<typename Self, NodeKind K, TypeKind T>
```
Maybe we can just assume Primitive implies a single return type.

ezyang commented on torch/csrc/jit/ir.h:
```diff
@@ -89,21 +194,26 @@ struct Node {
+  const Type* type() {
+    return type_.get();
+  }
+  void inferTypeFrom(const at::Tensor& output) {
```
This method seems like a layering violation: the IR header shouldn't know anything about ATen tensors.

ezyang commented on torch/csrc/jit/ir.h:
```diff
@@ -30,7 +33,109 @@ struct Graph;
+  friend struct Type;
+
+  std::vector<std::int64_t> sizes_;
+  std::vector<std::int64_t> strides_;
```
Any reason the actual type (e.g., float, double, etc.) isn't also recorded here?

ezyang committed:
Generalize update-xla-commit-hash and setup torchdynamo hash

The hash isn't used for anything yet.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

ghstack-source-id: 1d95b65cd9d99f4b83c2a2b0f74457fe81d9f8ad
Pull Request resolved: https://github.com/pytorch/pytorch/pull/79042

pytorchmergebot committed:
update xla commit hash
